Client/mods/deathmatch/logic/luadefs/CLuaDiscordDefs.cpp

```diff
@@ -0,0 +1,172 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include <lua/CLuaFunctionParser.h>
+
+void CLuaDiscordDefs::LoadFunctions()
+{
+    // Backwards compatibility functions
+    constexpr static const std::pair<const char*, lua_CFunction> functions[]{
+        {"setDiscordApplicationID", ArgumentParser<SetAppID>},
+        {"setDiscordRichPresenceDetails", ArgumentParser<SetDetails>},
+        {"setDiscordRichPresenceState", ArgumentParser<SetState>},
+        {"setDiscordRichPresenceAsset", ArgumentParser<SetLargeAsset>},
+        {"setDiscordRichPresenceSmallAsset", ArgumentParser<SetSmallAsset>},
+        {"setDiscordRichPresenceButton", ArgumentParser<SetButtons>},
+        {"resetDiscordRichPresenceData", ArgumentParser<ResetData>},
+        {"isDiscordRichPresenceConnected", ArgumentParser < IsDiscordRPCConnected>},
+
+    };
+
+    // Add functions
+    for (const auto& [name, func] : functions)
+        CLuaCFunctions::AddFunction(name, func);
+}
+
+void CLuaDiscordDefs::AddClass(lua_State* luaVM)
+{
+    lua_newclass(luaVM);
+
+    lua_classfunction(luaVM, "setApplication", "setDiscordApplicationID");
+    lua_classfunction(luaVM, "setState", "setDiscordRichPresenceState");
+    lua_classfunction(luaVM, "setDetails", "setDiscordRichPresenceDetails");
+    lua_classfunction(luaVM, "setAsset", "setDiscordRichPresenceAsset");
+    lua_classfunction(luaVM, "setSmallAsset", "setDiscordRichPresenceSmallAsset");
+    lua_classfunction(luaVM, "setButton", "setDiscordRichPresenceButton");
+
+    lua_classfunction(luaVM, "isConnected", "isDiscordRichPresenceConnected");
+    //lua_classfunction(luaVM, "setAppID", "setDiscordRichPresenceAppID");
+    //lua_classfunction(luaVM, "setAppID", "setDiscordRichPresenceAppID");
+
+
+    lua_registerclass(luaVM, "DiscordRPC");
+}
+
+bool CLuaDiscordDefs::ResetData()
+{
+    auto discord = g_pCore->GetDiscord();
+
+    if (!discord || !discord->IsDiscordRPCEnabled() || !discord->ResetDiscordData())
+        return false;
+
+    return true;
+}
+
+bool CLuaDiscordDefs::SetState(std::string strState)
+{
+    int stateLength = strState.length();
+
+    if (stateLength < 1 || stateLength > 128)
+        throw std::invalid_argument("State name must be greater than 0, or less than/equal to 128");
+
+    auto discord = g_pCore->GetDiscord();
+
+    if (!discord || !discord->IsDiscordRPCEnabled() || !discord->SetPresenceState(strState.c_str(), true))
+        return false;
+
+    return true;
+}
+
+bool CLuaDiscordDefs::SetAppID(std::string strAppID)
+{
+    int appIDLength = strAppID.length();
+
+    if (appIDLength > 32)
```

Comment:
Need to check for `appIDLength < 1` here

Reply:
> Need to check for `appIDLength < 1` here

Discord APPids are at least 17 characters long

Reply:
> > Need to check for `appIDLength < 1` here
> 
> Discord APPids are at least 17 characters long

That's exactly why you need to check if its `< 17` or empty (`< 1`)

Reply:
Either way, the expression needs to match the invalid argument exception. Either one is fine but I would prefer not to be bound to limits which Discord could change, so a minimum of `1` character is preferable here.

Reply:
As for the number of characters in the application ID, it varies, sometimes it is 17, 18 or even 20. So I would stick with 32.